pkg/segment/writer/segstore.go

```diff
@@ -1506,15 +1506,32 @@ func writeSstToBuf(sst *structs.SegStats, buf []byte) (uint16, error) {
 	copy(buf[idx:], toputils.Uint64ToBytesLittleEndian(sst.Count))
 	idx += 8
 
-	hllData := sst.GetHllBytes()
+	hllDataSize := sst.GetHllDataSize()
```

Comment:
this is actually pre-existing issue. `idx` is `uint16` so max val it can hold is 65535. However when we encode hll and other segstats, and if we change some of the hllDefaultSettings , it will cross 65K and we will have an overflow in the buf. (buf can hold upto 2 MB). 
But then the other issue is this version of this sst file allocates only 2 bytes for the hll size, so we will have to change that as well.